src/modules/orbitdb/abstract.js

```diff
@@ -27,18 +28,31 @@ class ChluAbstractIndex {
                             multihash: item.payload.multihash,
                             bitcoinTransactionHash: item.payload.bitcoinTransactionHash
                         });
-                    } else if (item.payload.op === operations.UPDATE_REVIEW_RECORD && IPFSUtils.isValidMultihash(item.payload.multihash) && IPFSUtils.isValidMultihash(item.payload.previousVersionMultihash)) {
-                        // TODO: check that whatever is being updated is in the DB
-                        // the multihash to be updated might already be an update. Be careful!
-                        const from = this.getLatestReviewRecordUpdate(item.payload.previousVersionMultihash);
-                        this.addReviewRecordUpdate({
-                            fromMultihash: from,
-                            toMultihash: item.payload.multihash
-                        });
+                    } else if (item.payload.op === operations.UPDATE_REVIEW_RECORD) {
+                        if (IPFSUtils.isValidMultihash(item.payload.multihash) && IPFSUtils.isValidMultihash(item.payload.previousVersionMultihash)) {
+                            // TODO: check that whatever is being updated is in the DB
+                            // the multihash to be updated might already be an update. Be careful!
+                            const from = this.getLatestReviewRecordUpdate(item.payload.previousVersionMultihash);
+                            this.addReviewRecordUpdate({
+                                fromMultihash: from,
+                                toMultihash: item.payload.multihash
+                            });
+                        }
+                    } else if (item.payload.op === operations.PUT_DID) {
+                        if (DID.isDIDID(item.payload.didId) && IPFSUtils.isValidMultihash(item.payload.multihash)) {
+                            // TODO: check signature
+                            this.putDID(item.payload.didId, item.payload.multihash, item.payload.signature)
+                        }
+                    } else if (item.payload.op === operations.PUT_UNVERIFIED_REVIEWS) {
+                        if (DID.isDIDID(item.payload.didId) && IPFSUtils.isValidMultihash(item.payload.reviewsMultihash)) {
```

Comment:
Just looking at this, I am thinking won't we have to run the same checks for DID when adding a review record?

Reply:
At the moment the code does not care (beyond simple checks like these) that an operation is valid before indexing the results into the OrbitDB Index.

Eventually, this `updateIndex` will be async so that we can run validations for ReviewRecords, and invalid review records should then never reach the index. This has the consequence that we won't be able to get a list of invalid data in the network though.

The DID is checked in the validation stage when you read a review record, by the same code that checks that the RR is properly signed by a DID.